lib/apiconfig/load.go

```diff
@@ -75,3 +80,23 @@ func LoadClientConfigFromEnvironment() (*CalicoAPIConfig, error) {
 
 	return c, nil
 }
+
+// ClientConfigOverride changes datastore type to Kubernetes when EtcdV3 endpoint is not
+// explicitly defined by user.
+func ClientConfigOverride(cfg *CalicoAPIConfig) (*CalicoAPIConfig, error) {
+	if cfg.Spec.EtcdEndpoints == "" && cfg.Spec.DatastoreType == EtcdV3 {
```

Comment:
I feel like checking `EtcdEndpoints` is enough, it can prevent an error in case user set `Kubernetes` as `datastoretype` but forgets to set `K8sAPIEndpoint` or `Kubeconfig`.

@robbrockbank @caseydavenport